spec/lib/appsignal/check_in_spec.rb

```diff
@@ -0,0 +1,303 @@
+          expect(Appsignal::Heartbeat).to be(Appsignal::CheckIn::Cron)
+        end
+      end
+
+    expect(logs).not_to contains_log(
+      :warn,
+      "The constant Appsignal::Heartbeat has been deprecated."
+    )
+  end
+end
+
+describe "Appsignal.heartbeat" do
+  let(:err_stream) { std_stream }
+
+  before do
+    Appsignal.instance_variable_set(:@heartbeat_helper_deprecation_warning_emitted, false)
+  end
+
+  it "should forward the call to Appsignal::CheckIn.cron" do
+    expect(Appsignal::CheckIn).to receive(:cron).with("heartbeat-name")
+    expect do
+      Appsignal.heartbeat("heartbeat-name")
+    end.not_to raise_error
+
+    block = proc { 42 }
+    expect(Appsignal::CheckIn).to receive(:cron).with("heartbeat-name") do |&given_block|
+      expect(given_block).to be(block)
+    end.and_return("output")
+    expect(Appsignal.heartbeat("heartbeat-name", &block)).to eq("output")
+  end
+
+  it "prints a deprecation warning to STDERR" do
+    capture_std_streams(std_stream, err_stream) do
+      Appsignal.heartbeat("heartbeat-name")
+    end
+
+    expect(err_stream.read)
+      .to include("appsignal WARNING: The helper Appsignal.heartbeat has been deprecated.")
+  end
+
+  it "does not print a deprecation warning to STDERR more than once" do
+    capture_std_streams(std_stream, err_stream) do
+      Appsignal.heartbeat("heartbeat-name")
+    end
+
+    expect(err_stream.read)
+      .to include("appsignal WARNING: The helper Appsignal.heartbeat has been deprecated.")
+
+    err_stream.truncate(0)
+
+    capture_std_streams(std_stream, err_stream) do
+      Appsignal.heartbeat("heartbeat-name")
+    end
+
+    expect(err_stream.read)
+      .not_to include("appsignal WARNING: The helper Appsignal.heartbeat has been deprecated.")
+  end
+
+  it "logs a warning" do
+    logs =
+      capture_logs do
+        silence do
+          Appsignal.heartbeat("heartbeat-name")
+        end
+      end
+
+    expect(logs).to contains_log(
+      :warn,
+      "The helper Appsignal.heartbeat has been deprecated."
+    )
+  end
+
+  it "does not log a warning more than once" do
+    logs =
+      capture_logs do
+        silence do
+          Appsignal.heartbeat("heartbeat-name")
+        end
+      end
+
+    expect(logs).to contains_log(
+      :warn,
+      "The helper Appsignal.heartbeat has been deprecated."
+    )
+
+    logs =
+      capture_logs do
+        silence do
+          Appsignal.heartbeat("heartbeat-name")
+        end
+      end
+
+    expect(logs).not_to contains_log(
+      :warn,
+      "The helper Appsignal.heartbeat has been deprecated."
+    )
+  end
+end
+
+describe Appsignal::CheckIn::Cron do
+  let(:config) { project_fixture_config }
+  let(:cron_checkin) { described_class.new(:name => "cron-checkin-name") }
+  let(:transmitter) { Appsignal::Transmitter.new("http://cron_checkins/", config) }
+
+  before(:each) do
+    allow(Appsignal).to receive(:active?).and_return(true)
+    config.logger = Logger.new(StringIO.new)
```

Comment:
Nit: I think this is unchanged from the heartbeat spec, but it would be nice to update this to use `start_agent` and `capture_logs`, after this PR is merged. I can pick this up.

Reply:
Appreciated, feel free to pick it up after the PR is merged.